service/service_test.go

```diff
@@ -269,7 +269,9 @@ func testCollectorStartHelper(t *testing.T, useOtel bool, tc ownMetricsTestCase)
 		// Sleep for 1 second to ensure the http server is started.
 		time.Sleep(1 * time.Second)
 		assert.True(t, loggingHookCalled)
-		assertMetrics(t, metricsAddr, tc.expectedLabels)
+		if !useOtel {
```

Comment:
Should assertMetrics not work regardless of whether it's using otel or not?

Reply:
If I'm understanding `assertMetrics` correctly, it is meant to check that certain labels are present from the metrics found at the metrics endpoint.  Now that otel-go instrumentation is used when featuregate is enabled, these labels do not apply to otel.  OCMetrics however remain unchanged, so they will have the expected labels.

Reply:
For reference:
*featuregate enabled* only a few of the metrics have the expected labels like `service_instance_id`
```
% curl localhost:8888/metrics                         
# HELP otelcol_exporter_enqueue_failed_log_records Number of log records failed to be added to the sending queue.
# TYPE otelcol_exporter_enqueue_failed_log_records counter
otelcol_exporter_enqueue_failed_log_records{exporter="logging",service_instance_id="54494d1d-76a1-42e2-a128-9bdbb0668396",service_name="otelcorecol",service_version="0.68.0-dev"} 0
# HELP otelcol_exporter_enqueue_failed_metric_points Number of metric points failed to be added to the sending queue.
# TYPE otelcol_exporter_enqueue_failed_metric_points counter
otelcol_exporter_enqueue_failed_metric_points{exporter="logging",service_instance_id="54494d1d-76a1-42e2-a128-9bdbb0668396",service_name="otelcorecol",service_version="0.68.0-dev"} 0
# HELP otelcol_exporter_enqueue_failed_spans Number of spans failed to be added to the sending queue.
# TYPE otelcol_exporter_enqueue_failed_spans counter
otelcol_exporter_enqueue_failed_spans{exporter="logging",service_instance_id="54494d1d-76a1-42e2-a128-9bdbb0668396",service_name="otelcorecol",service_version="0.68.0-dev"} 0
# HELP otelcol_process_cpu_seconds_total Total CPU user and system time in seconds
# TYPE otelcol_process_cpu_seconds_total counter
otelcol_process_cpu_seconds_total 0.06
# HELP otelcol_process_memory_rss Total physical memory (resident set size)
# TYPE otelcol_process_memory_rss gauge
otelcol_process_memory_rss 3.432448e+07
# HELP otelcol_process_runtime_heap_alloc_bytes Bytes of allocated heap objects (see 'go doc runtime.MemStats.HeapAlloc')
# TYPE otelcol_process_runtime_heap_alloc_bytes gauge
otelcol_process_runtime_heap_alloc_bytes 5.4054712e+08
# HELP otelcol_process_runtime_total_alloc_bytes_total Cumulative bytes allocated for heap objects (see 'go doc runtime.MemStats.TotalAlloc')
# TYPE otelcol_process_runtime_total_alloc_bytes_total counter
otelcol_process_runtime_total_alloc_bytes_total 5.42846208e+08
# HELP otelcol_process_runtime_total_sys_memory_bytes Total bytes of memory obtained from the OS (see 'go doc runtime.MemStats.Sys')
# TYPE otelcol_process_runtime_total_sys_memory_bytes gauge
otelcol_process_runtime_total_sys_memory_bytes 5.77772688e+08
# HELP otelcol_process_uptime_total Uptime of the process
# TYPE otelcol_process_uptime_total counter
otelcol_process_uptime_total 10.014383
# HELP otelcol_target_info Target metadata
# TYPE otelcol_target_info gauge
otelcol_target_info{service_instance_id="54494d1d-76a1-42e2-a128-9bdbb0668396",service_name="otelcorecol",service_version="0.68.0-dev"} 1
```

Reply:
*featuregate disabled* (OC) all the metrics seem to have the labels that assertMetrics is expecting
```
% curl localhost:8888/metrics   
# HELP otelcol_exporter_enqueue_failed_log_records Number of log records failed to be added to the sending queue.
# TYPE otelcol_exporter_enqueue_failed_log_records counter
otelcol_exporter_enqueue_failed_log_records{exporter="logging",service_instance_id="d86342f8-6b14-4451-afdc-9fcd60b7aee8",service_name="otelcorecol",service_version="0.68.0-dev"} 0
# HELP otelcol_exporter_enqueue_failed_metric_points Number of metric points failed to be added to the sending queue.
# TYPE otelcol_exporter_enqueue_failed_metric_points counter
otelcol_exporter_enqueue_failed_metric_points{exporter="logging",service_instance_id="d86342f8-6b14-4451-afdc-9fcd60b7aee8",service_name="otelcorecol",service_version="0.68.0-dev"} 0
# HELP otelcol_exporter_enqueue_failed_spans Number of spans failed to be added to the sending queue.
# TYPE otelcol_exporter_enqueue_failed_spans counter
otelcol_exporter_enqueue_failed_spans{exporter="logging",service_instance_id="d86342f8-6b14-4451-afdc-9fcd60b7aee8",service_name="otelcorecol",service_version="0.68.0-dev"} 0
# HELP otelcol_process_cpu_seconds Total CPU user and system time in seconds
# TYPE otelcol_process_cpu_seconds counter
otelcol_process_cpu_seconds{service_instance_id="d86342f8-6b14-4451-afdc-9fcd60b7aee8",service_name="otelcorecol",service_version="0.68.0-dev"} 0.08
# HELP otelcol_process_memory_rss Total physical memory (resident set size)
# TYPE otelcol_process_memory_rss gauge
otelcol_process_memory_rss{service_instance_id="d86342f8-6b14-4451-afdc-9fcd60b7aee8",service_name="otelcorecol",service_version="0.68.0-dev"} 3.426304e+07
# HELP otelcol_process_runtime_heap_alloc_bytes Bytes of allocated heap objects (see 'go doc runtime.MemStats.HeapAlloc')
# TYPE otelcol_process_runtime_heap_alloc_bytes gauge
otelcol_process_runtime_heap_alloc_bytes{service_instance_id="d86342f8-6b14-4451-afdc-9fcd60b7aee8",service_name="otelcorecol",service_version="0.68.0-dev"} 5.4050804e+08
# HELP otelcol_process_runtime_total_alloc_bytes Cumulative bytes allocated for heap objects (see 'go doc runtime.MemStats.TotalAlloc')
# TYPE otelcol_process_runtime_total_alloc_bytes counter
otelcol_process_runtime_total_alloc_bytes{service_instance_id="d86342f8-6b14-4451-afdc-9fcd60b7aee8",service_name="otelcorecol",service_version="0.68.0-dev"} 5.4280048e+08
# HELP otelcol_process_runtime_total_sys_memory_bytes Total bytes of memory obtained from the OS (see 'go doc runtime.MemStats.Sys')
# TYPE otelcol_process_runtime_total_sys_memory_bytes gauge
otelcol_process_runtime_total_sys_memory_bytes{service_instance_id="d86342f8-6b14-4451-afdc-9fcd60b7aee8",service_name="otelcorecol",service_version="0.68.0-dev"} 5.77772688e+08
# HELP otelcol_process_uptime Uptime of the process
# TYPE otelcol_process_uptime counter
otelcol_process_uptime{service_instance_id="d86342f8-6b14-4451-afdc-9fcd60b7aee8",service_name="otelcorecol",service_version="0.68.0-dev"} 3.199575
```

Reply:
Do you think I should manually enrich the proctelemetry.ProcessMetrics with these `service_*` labels by adding the otelattrs? Is there a way to retrieve the values of these labels to add them to otel, or should these not apply anymore?

Reply:
I would expect those labels to process metrics as they do for exporter metric, see `otelcol_exporter_enqueue_failed_log_records` for example.

Reply:
@codeboten Hmm but isn't `otelcol_exporter_enqueue_failed_log_records` an OC instrument and not an Otel one?  I think so based off this [file](https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/exporterhelper/obsreport.go#L78).  So I would expect it to have the `service_*` labels.  I don't think otel provides these same labels. Is there any way we can confirm this?

Reply:
@codeboten So after further discussion with Jmac,  it seems resource attributes are created during [telemetry initialization](https://github.com/open-telemetry/opentelemetry-collector/blob/e5208293ec5d4e04939ff52d60519ddbaa12d87a/service/telemetry.go#L93) and passed to both OC and Otel initialization functions.

This helped realize that these attributes actually aren't missing at all, but instead recorded in one place for otel - the `target_info` metric.  According to otel specification about resource attributes 

```
In SDK Prometheus (pull) exporters, resource attributes SHOULD be converted to a single target_info metric; otherwise, they MUST be dropped, and MUST NOT be attached as labels to other metric families.
```

So the resource attributes are simply captured in the `otel_target_info` resource attributes (it is not necessary to have these attributes repeat in the other metrics). So I think the sdk exposes these attributes once in the target_info metric through the prometheus exporter. And therefore these attributes can be recovered for all the otel metrics at query-time from this `target_info`.  

cc @jmacd in case I'm missing any important info

Reply:
LGTM thanks @moh-osman3

Reply:
Thanks @moh-osman3, now that I look at `assertMetrics` more closely I realize that my confusion stemmed from thinking this was performing the same type of test as the `assertMetrics` in the batchprocessor tests was doing. Seeing the output above with metrics emitted from both OC and OT further confused me. Thank you for clarifying. 

A follow up to this would be to change `assertMetrics` to either validate the metrics themselves, or rename it to `assertOCLabels` (or something like that).

Reply:
Thanks for the review @codeboten !  Your questions were helpful to improve my understanding as well. I created this issue https://github.com/open-telemetry/opentelemetry-collector/issues/7075 to track your suggested improvements on the `assertMetrics` function.